### ISSUE TITLE
FR: page labels for DjVu

### ISSUE THREAD

**Issue by Frenzie** [koreader/koreader#3833]
* KOReader version:  38a0519 
* Device: all

#### Issue
This is nice to have, but I don't immediately see how to get it out of any of the functions in ddjuapi.h. For looking into those miniexp Lisp things use `miniexp_prin()`.

![screenshot_2018-04-05_20-04-18](https://user-images.githubusercontent.com/202757/38383339-92888f44-390c-11e8-88a0-75f67bbdbd1b.png)

Context [here](https://www.mobileread.com/forums/showthread.php?p=3678402#post3678402).

>Assuming you're talking about custom page numbers, that doesn't seem to be in the MuPDF API.
>
>Spec for reference: http://pdf.editme.com/pdfua-PageLabels
>
>Upstream bug here: https://bugs.ghostscript.com/show_bug.cgi?id=695351 (branch that patch was developed on [here](https://github.com/asbloomf/mupdf/tree/page-labels-1.9-squash)).

Pinging @cramoisi Is this also a thing in EPUBs?

**Comment by cramoisi** [comment]
yep, I never saw that as a problem though. I really never have the need to have my epub copy the page of the paperback edition - or even indicate them. (I just want to know the number of pages I've yet to read and that's all). Although, there is some code to mimic the "paperback pages" (need to find a book with that, i've seen some spans id as pageN here and there but I'm not even sure RMSDK or kindle render them). But again, that doesn't make sense to me since each device and config make the pagination different. It would be important for citations in a paper but I wouldn't dare quoting  Descartes'meditations in a epub edition anyway, nor anything in epub format.

Also, this could be endless : there is a lot of books with two or three sets of pagination (like one in roman char for the introduction, etc. ; you know that with gallica's pdf).

Anyway, ktime9 on mobileread is right for a thing : according to the library's science the cover, le deuxième plat, le troisième plat (don't know the words for these two, they are the verso of the cover and the recto of the back cover) and the back cover are not "pages".

(and according to me epub should get rid of inner TOC since they are remnants of a paper's way of thinking as is the "cover" : it doesn't really exist since there is no back-cover. It's just metadata.)

**Comment by Frenzie** [comment]
I just meant I want it displayed, not to edit it. (Why would you even think of such complications. :-P) Now you'd think the easy part would be getting the info *from* djvulibre but I can't seem to find it. There's a code path that makes reference to it but it doesn't lead to non-numerical page labels.

Anyway, for EPUB it may not be too hard either then. For MuPDF I don't want to depend on a custom patch though. :-)

**Comment by cramoisi** [comment]
(it was late... sorry)
For epub It would be easy to enable for anyone who want it if publishers follows the same rules with an inline tag (which is not really the case). But I would defeat a great feat of koreader which can tells to real page number (aka the number of time we have to touch the screen for switching the page). it's not bad to let people choose anyway.

For now, I found some examples comparing books I have in both paper/epub official format : 

`<span><a id="page_11"></a>1</span>` (at the beginning of each paperback pages for the first french edition)
`<div id="page1">`....</span>` (around each content of the text of the first edition).
`<span id="page1" />` (at the beginning of the text)

As I'm a guy who sometime write texts with a lot of references I only see problems here: because it's made to mimic the paper, we could quote the ebook and the ref would be right for the paper, but we don't know which paperback edition it mimics (it's not mentioned), so the best would be to quote the ebook (something nobody would do). So for the writing aspect, it's useless

Where it would be great is when dual reading the same book one in paperback and the other in epub (as I often do with my wife), or anything which would mix paper/epub. But again, it would only works if the epub follows _this_ paperback edition. For personal use it can't think of anything which would be of use.

But, as a librarian, I would FR the fact that the "cover" to be removed of the count of pages  ;-D

**Comment by Frenzie** [comment]
@poire-z I can't find the relevant issue atm, but I can reliably reproduce the tearing issue with my label test file. (It doubles as a Kobo Cyrillic DjVu crash test file. Actually I haven't done any label stuff at this point in time yet other than some cursory glances at ddjvuapi.h.)

[Михайлов ТФКП.djvu.zip](https://github.com/koreader/koreader/files/1886295/djvu.zip)

[2018-04-07 12.45.02.mp4.zip](https://github.com/koreader/koreader/files/1886298/2018-04-07.12.45.02.mp4.zip)

**Comment by Frenzie** [comment]
Pinging @xelxebar

**Comment by xelxebar** [comment]
@Frenzie 
Thanks for the ping. I'll look into this as well.

**Comment by Frenzie** [comment]
Cross-ref to #4245.
